src/errors.ts

```diff
@@ -0,0 +1,16 @@
+export class WhatsABIError extends Error {
+    override name = 'WhatsABIError';
+
+    // Some variables included from the context scope of the error, for debugging
+    context?: Record<string, any>;
+
+    constructor(message: string, args: { context?: Record<string, any>, cause?: Error } = {}) {
+        super(message, { cause: args.cause } as ErrorOptions);
+
+        this.context = args.context;
+    }
+}
+
+export class AutoloadError extends WhatsABIError {}
```

Comment:
```suggestion
export class AutoloadError extends WhatsABIError {
  override name AutoloadError = 'AutoloadError' as const
}
```
So the reason why giving a name and using as const like this is useful is because it fixes a corner case bug of using instanceof.

So we could descriminate these errors using instanceof

```typescript
import {AutoLoadError} from 'shazow/whatsabi'

if (err instanceof AutoLoadError) {
  handleIt()
}
```

The issue of doing this is there are some corner cases where this won't work. Namely, if more than one version of whatsabi for some reason ended up installed. If more than one version of whatsabi are installed it can be the same error type but not actually the same instanceof. For this reason viem always makes `name` property strongly typed and in documentation points end users to using name property to descriminate errors

Reply:
Fair, will make that change!